### PR TITLE
SOV-2294: Pull Zero fee rates directly from contracts - reimplementation

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.atlassian.net/browse/SOV-2295

### CODE REVIEW

File: apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts [hunk 1]
```diff
@@ -7,11 +7,53 @@ import {
...
 
-const defaultBorrowingRateSlippageTolerance = 0.005; // 0.5%
+interface ILiquityBaseParams {
+  minBorrowingFeeRate: UtilsDecimal;
+  maxBorrowingFeeRate: UtilsDecimal;
+}
+
+let cachedParams: ILiquityBaseParams | undefined;
+
+export const getLiquityBaseParams = async (): Promise<ILiquityBaseParams> => {
```
Comment (Rickk137) [inline apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts]: I think this function should not be in this file. Maybe create another file?

File: apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts [hunk 2]
```diff
@@ -7,11 +7,53 @@ import {
...
+    );
+
+    const params = {
+      minBorrowingFeeRate,
+      maxBorrowingFeeRate,
+    };
+
+    cachedParams = params;
+
+    return params;
```
Comment (Rickk137) [inline apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts]:
```
cachedParams = {
      minBorrowingFeeRate,
      maxBorrowingFeeRate,
    };
    return cachedParams;
```

**Review by Rickk137 — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/5_pages/ZeroPage/hooks/useLiquityBaseParams.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react';
+
+import { Decimal } from '@sovryn/utils';
+
+import { getLiquityBaseParams } from '../utils/trove-manager';
+
+export const useLiquityBaseParams = () => {
+  const [liquityBaseParams, setLiquityBaseParams] = useState({
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/ZeroPage/hooks/useLiquityBaseParams.ts]:
```suggestion
  const [liquityBaseParams, setLiquityBaseParams] = useState<<ILiquityBaseParams>>({
```
`minBorrowingFeeRate` and `maxBorrowingFeeRate` have an incorrect type (`any`) without this.

File: apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts (hunk 1 above)
Comment (tiltom) [reply apps/frontend/src/app/5_pages/ZeroPage/utils/trove-manager.ts]: Exactly, this is not the place for this function. If you won't figure out a better location, I'd propose creating `index.ts` file in the current folder (`utils`) and placing it there.

**Review by tiltom — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/5_pages/ZeroPage/types.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import { Decimal } from '@sovryn/utils';
+
+export interface ILiquityBaseParams {
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/ZeroPage/types.ts]:
```suggestion
export type LiquityBaseParams {
```
Always use `type` if possible.

File: apps/frontend/src/app/5_pages/ZeroPage/utils/index.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+      maxBorrowingFee.toString(),
+    );
+
+    cachedParams = {
+      minBorrowingFeeRate,
+      maxBorrowingFeeRate,
+    };
+    return cachedParams;
+  } catch (error) {
+    console.error(error);
```
Comment (tiltom) [inline apps/frontend/src/app/5_pages/ZeroPage/utils/index.ts]:
```suggestion

```

**Review by tiltom — CHANGES_REQUESTED** [review]

**Review by tiltom — APPROVED** [review]

**Pull request merged**